driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -231,7 +231,7 @@ void maybeAddRoutingKey(String name, Object value) {
             return;
 
         for (int i = 0; i < partitionKey.size(); i++) {
-            if (name.equals(partitionKey.get(i).getName())) {
+            if (name.equalsIgnoreCase(partitionKey.get(i).getName())) {
```

Comment:
Great point, I didn't even consider that `Metadata.handleId` wasn't public.  I'm not sure what the best direction is, but I think it seems reasonable to me to make it public although there could be reasons to not make it so.  @olim7t / @adutra / @newkek, thoughts?